test/e2e/virtual/initializingworkspaces/virtualworkspace_test.go

```diff
@@ -181,26 +198,74 @@ func TestInitializingWorkspacesVirtualWorkspaceAccess(t *testing.T) {
-		virtualWorkspaceRawConfig.Clusters[initializer].Server = adminCluster.Server + "/services/initializingworkspaces/" + string(initializerName)
-		virtualWorkspaceRawConfig.Contexts[initializer] = adminContext.DeepCopy()
-		virtualWorkspaceRawConfig.Contexts[initializer].Cluster = initializer
-		virtualWorkspaceConfig, err := clientcmd.NewNonInteractiveClientConfig(*virtualWorkspaceRawConfig, initializer, nil, nil).ClientConfig()
-		require.NoError(t, err)
-		virtualKcpClusterClient, err := kcpclient.NewClusterForConfig(virtualWorkspaceConfig)
+		virtualWorkspaceConfig.Host = baseURL + string(initializerName)
+		virtualKcpClusterClient, err := kcpclient.NewClusterForConfig(userConfig("user-1", virtualWorkspaceConfig))
 		require.NoError(t, err)
 		clients[initializer] = virtualKcpClusterClient.Cluster(logicalcluster.Wildcard).TenancyV1alpha1().ClusterWorkspaces()
 	}
 
+	t.Log("Ensure that LIST calls through the virtual workspace fail authorization")
+	for _, initializer := range []string{
+		"alpha",
+	} {
+		_, err := clients[initializer].List(ctx, metav1.ListOptions{})
+		if !errors.IsForbidden(err) {
+			t.Fatalf("got %#v error from initial list, expected unauthorized", err)
+		}
+	}
+
+	t.Log("Set up RBAC to allow future calls to succeed")
+	for _, initializer := range []string{
+		"alpha",
+	} {
+		cwt := clusterWorkspaceTypes[initializer]
+		role, err := kubeClusterClient.Cluster(clusterName).RbacV1().ClusterRoles().Create(ctx, &rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: string(initialization.InitializerForType(cwt)) + "-initializer",
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					Verbs:         []string{"initialize"},
+					Resources:     []string{"clusterworkspacetypes"},
+					ResourceNames: []string{cwt.Name},
+					APIGroups:     []string{"tenancy.kcp.dev"},
+				},
+			},
+		}, metav1.CreateOptions{})
+		require.NoError(t, err)
+		source.Artifact(t, func() (runtime.Object, error) {
+			return kubeClusterClient.Cluster(clusterName).RbacV1().ClusterRoles().Get(ctx, role.Name, metav1.GetOptions{})
+		})
+		binding, err := kubeClusterClient.Cluster(clusterName).RbacV1().ClusterRoleBindings().Create(ctx, &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: role.Name,
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind:     "ClusterRole",
+				APIGroup: "rbac.authorization.k8s.io",
+				Name:     role.Name,
+			},
+			Subjects: []rbacv1.Subject{{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "User",
+				Name:     "user-1",
+			}},
+		}, metav1.CreateOptions{})
+		require.NoError(t, err)
+		source.Artifact(t, func() (runtime.Object, error) {
+			return kubeClusterClient.Cluster(clusterName).RbacV1().ClusterRoleBindings().Get(ctx, binding.Name, metav1.GetOptions{})
+		})
+	}
+
 	t.Log("Ensure that LIST calls through the virtual workspace show the correct values")
 	workspaces, err := sourceKcpTenancyClient.ClusterWorkspaces().List(ctx, metav1.ListOptions{
```

Comment:
this need require.Eventually now as authz is async (full of informers).

Reply:
TODO: also do this in other VW tests

Reply:
Created follow-up task in the epic?